internal/thelma/toolbox/argocd/argocd.go

```diff
@@ -137,7 +134,7 @@ type argocdConfig struct {
 	WaitInProgressOperationTimeoutSeconds int `default:"300"`
 
 	// SyncTimeoutSeconds timeout for sync operations
-	SyncTimeoutSeconds int `default:"600"`
+	SyncTimeoutSeconds int `default:"900"`
```

Comment:
Would it make sense to see if retries help first before bumping the overall timeout? Otherwise it might be difficult to tell which actually alleviated the issue.

Reply:
So the intention behind these changes is that if a sync or wait-healthy fails with a timeout error (`Context Canceled`), we don't retry. We already waited 900 seconds. Something is really broken, we should bubble the error up.

But the more aggressive retry-by-default approach should catch the random / intermittent 500 errors we've been seeing.

Reply:
Makes sense!